From a5e27e95e6477d750bfe1a82bde56c3971ff10e5 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Sun, 8 Jun 2014 15:58:47 +1000
Subject: [PATCH] Make MythCoreContext::GetMasterServerIP returns localhost
 when none are defined

(cherry picked from commit c0223f9ffa35b4d16768e96aa989b7f1998b5610)
---
 mythtv/libs/libmythbase/mythcorecontext.cpp |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythbase/mythcorecontext.cpp b/mythtv/libs/libmythbase/mythcorecontext.cpp
index 11e1519..b5a4822 100644
--- a/mythtv/libs/libmythbase/mythcorecontext.cpp
+++ b/mythtv/libs/libmythbase/mythcorecontext.cpp
@@ -861,7 +861,9 @@ double MythCoreContext::GetFloatSettingOnHost(const QString &key,
  */
 QString MythCoreContext::GetMasterServerIP(void)
 {
-    return resolveSettingAddress("MasterServerIP");
+    QString master = resolveSettingAddress("MasterServerIP");
+
+    return master.isEmpty() ? "localhost" : master;
 }
 
 /**
-- 
1.7.10.2

